tensorzero-core/src/db/clickhouse/migration_manager/migrations/migration_0036.rs

```diff
@@ -0,0 +1,211 @@
+            FROM ModelInference
+            {view_where_clause}
+            GROUP BY (model_name, model_provider_name, minute)
+            "
+        );
+        self.clickhouse
+            .run_query_synchronous_no_params(query)
+            .await?;
+
+        // Backfill if needed
+        if !clean_start {
+            tokio::time::sleep(view_offset).await;
+
+            let create_table = self
+                .clickhouse
+                .run_query_synchronous_no_params(
+                    "SHOW CREATE TABLE ModelProviderStatisticsView".to_string(),
+                )
+                .await?
+                .response;
+
+            let view_timestamp_nanos_string = view_timestamp_nanos.to_string();
+            if !create_table.contains(&view_timestamp_nanos_string) {
+                tracing::warn!("Materialized view `ModelProviderStatisticsView` was not written because it was recently created. This is likely due to a concurrent migration. Unless the other migration failed, no action is required.");
+                return Ok(());
+            }
+
+            tracing::info!("Running backfill of ModelProviderStatistics");
+            let query = format!(
+                r"
+                INSERT INTO ModelProviderStatistics
+                SELECT
+                    model_name,
+                    model_provider_name,
+                    toStartOfMinute(timestamp) as minute,
+
+                    quantilesTDigestState({qs})(response_time_ms) as response_time_ms_quantiles,
+                    quantilesTDigestState({qs})(ttft_ms) as ttft_ms_quantiles,
+                    sumState(input_tokens) as total_input_tokens,
+                    sumState(output_tokens) as total_output_tokens,
+                    countState() as count
+                FROM ModelInference
+                WHERE UUIDv7ToDateTime(id) < fromUnixTimestamp64Nano({view_timestamp_nanos})
+                GROUP BY model_name, model_provider_name, minute
+                "
+            );
+            self.clickhouse
+                .run_query_synchronous_no_params(query)
+                .await?;
+        }
+
+        Ok(())
+    }
+
+    fn rollback_instructions(&self) -> String {
+        let on_cluster_name = self.clickhouse.get_on_cluster_name();
+        // N.B. The second line in your original looked like a typo.
+        // Keeping intent: drop the *table* that the MV targets.
+        format!(
+            r"
+        DROP TABLE IF EXISTS ModelProviderStatisticsView{on_cluster_name} SYNC;
+        DROP TABLE IF EXISTS ModelProviderStatistics{on_cluster_name} SYNC;"
+        )
+    }
+
+    async fn has_succeeded(&self) -> Result<bool, Error> {
+        let should_apply = self.should_apply().await?;
+        Ok(!should_apply)
+    }
+}
+
+/*
+Example query (using the same quantile list):
+SELECT
+    model_name,
+    model_provider_name,
+    minute,
+    quantilesTDigestMerge({qs})(response_time_ms_quantiles) AS response_time_quantiles,
```

Comment:
Anselmo says try other quantile functions ([QuantileDD](https://clickhouse.com/docs/sql-reference/aggregate-functions/reference/quantileddsketch), perhaps), quantile otherwise.